boards/arm/mr_canhubk3/mr_canhubk3_defconfig

```diff
@@ -18,6 +18,12 @@ CONFIG_ARM_MPU=y
 # Use no-cached memory for HAL
 CONFIG_NOCACHE_MEMORY=y
 
+# Use MDIO
+CONFIG_MDIO=y
+CONFIG_MDIO_SHELL=n
+CONFIG_MDIO_INIT_PRIORITY=81
```

Comment:
unfortunately the priorities must be tweaked, because the Ethernet GMAC driver will perform a soft reset of the hw block, also resetting MDIO. So the initialization order must be Ethernet driver and then MDIO driver.

Since this is SoC related, I suggest to move it directly to https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_s32/s32k3/Kconfig.defconfig.series#L25.
For example:
```Kconfig
# GMAC ethernet driver must be initialized before MDIO driver because
# because it will reset the GMAC hardware block including MDIO
config ETH_INIT_PRIORITY
       default 60

config MDIO_INIT_PRIORITY
       default 70

config PHY_INIT_PRIORITY
       default 80
```

Reply:
Great suggestion!